### ISSUE TITLE
W3C Validator errors

### ISSUE THREAD

**Issue by vtheod** [templaza/astroid-framework#311]
Hi all,

I was checking a demo site and w3c validator gives some errors. 
One error is about the Header Section Layout. If you choose the container-fluid section layout, astroid creates a div with class container-fluid, but this div remains unclosed (that div is created in the header section). 

Another error is that the main element must not appear as a descendant of the section element. The `<main class="astroid-component-area">` appears inside the `<section class="astroid-section astroid-component-section">`.

Can these errors be fixed?

If I find more errors, I'll post them here.

**Comment by Chacapamac** [comment]
Thank @vtheod 
I can confirm the error on my test website about the “main” —> http://joom4-astro.graf-test.ca/

———————  **Template One & Zero** ——————— 
• The **main** element must not appear as a descendant of the **section** element.
From line 255, column 69; to line 255, column 105

Maybe?
`<section id="cs-1528971561366" class="astroid-section astroid-component-section">`
Should be 
`<main id="cs-1528971561366" class="astroid-section astroid-component-section">`

and `<main class="astroid-component-area">`
Should be 
<section class="astroid-component-area">

**Not sure if possible?**


**I see another error:**
• Element [**dl**] is missing a required child element.

From line 272, column 16; to line 272, column 102
`          <dd class="createdby" itemprop="author" itemscope itemtype="https://schema.org/Person">`
Not sure about that one

**Comment by vtheod** [comment]
Thanks @Chacapamac 
Did you manage to reproduce the header section error too?

**Comment by Chacapamac** [comment]
No, I cannot see this error when the Header Section Layout settings is “Container Fluid”
![Screen Shot 2023-04-06 at 9 51 43 AM](https://user-images.githubusercontent.com/1907276/230399394-a5251156-b558-4587-9f37-ad6a1268ead4.jpg)

No unclosed div in both Template one and zero?

See http://joom4-astro.graf-test.ca/ — Template One —> Header Section Layout settings “Container Fluid”

**Comment by vtheod** [comment]
I checked it on another installation, you're right. Probably there's something on my end...

**Comment by Chacapamac** [comment]
@vtheod  & @sonvnn  what do you think of my solution for the W3C Validation error, 
> • The main element must not appear as a descendant of the section element.

**in my previous post** —> https://github.com/templaza/astroid-framework/issues/311#issuecomment-1499051889

It make sens to me, whenever you have an article showing in a page, it’s content become the “**main**” content. I think this the original intent of Joomla?

**Comment by vtheod** [comment]
I think that if we just switch the elements with each other, the error will be fixed

**Comment by sonvnn** [comment]
@Chacapamac It's so confuse because Main tag display for Component block only. It's use to Accessibility Check in Joomla
Let check this commit or cassiopeia template https://github.com/templaza/astroid-framework/commit/989dd4ad7de8bc9878b25799e806ca8202bc738c

The solution is replace all <section> tag to <div>.

I will think about it. Any ideas are welcome.

Best Regards,
Sonny
